### DISCUSSION TITLE
core-js issue: dependency was not found

### DISCUSSION THREAD

**Discussion by mjzhaochenyi** [quasarframework/quasar#9608] [CLI - Electron mode] [answered]
Hi Buddies,

I encountered a core-js issue, when I run "quasar dev". I searched a lot before I came to this forum. Ok. My error message is like below. All I did to my code is to try to use a for loop on an array, and then I get the error below.

The line that caused this problem is as below.
```
    Object.keys(state.tasks).forEach(function(key) {
      let task = state.tasks[key]
      if (!task.completed) {
        tasks[key] = task
      }
    })
```

The error is 
```
ERROR  Failed to compile with 1 error 
4:33:44 PM
This dependency was not found:

* core-js/core/array in ./src/store/store-tasks.js

To install it, you can run: npm install --save core-js/core/array
Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\hiberfil.sys'
```

My specs are as below  - run "quasar info"
```
PS C:\Users\Gaz C Zhao\Desktop\General\Education\Coding\Quasar\quasar-todo> quasar info

Operating System - Windows_NT(10.0.18363) - win32/x64
NodeJs - 14.15.5

Global packages 
  NPM - 7.14.0
  yarn - Not installed
  @quasar/cli - 1.2.0
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 1.15.20 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 2.2.9 -- Quasar Framework local CLI
  @quasar/extras - 1.10.6 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 2.6.13 -- Reactive, component-oriented view layer for modern web interfaces.
  vue-router - 3.5.1 -- Official router for Vue.js 2
  vuex - 3.6.2 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.14.3 -- Babel compiler core.
  webpack - 4.44.2 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 3.11.2 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.2.2 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - CNPF24XL2XZX
  Wi-Fi - 192.168.0.102
```

I tried every solution from the tread below, but none of them worked.
https://forum.quasar-framework.org/topic/5146/solved-core-js-issue/15
I removed node modules folder, I removed package-lock.json, and run npm install again. Did not work.
I tried to downgrade core-js to 2.6.5 version from 3.4. Did not work.
I added  { useBuiltIns: "entry" } to babel's config file. Did not work.

Currently stuck on this issue, can anyone provide an advice, please? Much appreciate.

-

**Comment by ontwikkelfabriek** [comment, marked as answer]
@mjzhaochenyi

looking at the output of your `quasar info`:

- use NodeJS v12.x  ( for Qv1 )
- use **yarn** for managing a Quasar project deps
- use npm for **global** deps/installs
- not sure about NPM 7.x , I would use 6.x

- remove node modules again ( because yarn should handle the quasar projects deps instead of npm)
- remove  package-lock.json ( because yarn should handle the quasar projects deps instead of npm)

> Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\hiberfil.sys'

I would reboot my PC to get rid of this message.

> C:\Users\Gaz C Zhao\Desktop\General\Education\Coding\Quasar\quasar-todo>

This is a long nested folder to put your quasar project.  I would place my projects in windows OS always under `c:\projects\myProject` You could run into windows folder limitations:

`There is an upper limit of 248 characters on the total path length to a file located on Windows Operating Systems. This includes nested folders. You may have as many nested folders as you wish, as long as the complete folder name does not exceed 248 characters.`

**Reply by mjzhaochenyi** [reply]
Hi @ontwikkelfabriek, thanks for the help. I tried the steps above. Still no luck. I will keep searching.

**Reply by mjzhaochenyi** [reply]
I solved it. Performing your steps indeed worked. Thanks.
